orchestrator/crates/orchestrator/src/core/client/database/mod.rs

```diff
@@ -59,4 +61,10 @@ pub trait DatabaseClient: Send + Sync {
         status: Vec<JobStatus>,
         limit: Option<i64>,
     ) -> Result<Vec<JobItem>, DatabaseError>;
+    /// get_latest_batch - Get the latest batch from DB. Returns `None` if the DB is empty
+    async fn get_latest_batch(&self) -> Result<Option<Batch>, DatabaseError>;
+    /// update_batch - Update the bath
+    async fn update_batch(&self, batch: &Batch, update: &BatchUpdates) -> Result<Batch, DatabaseError>;
+    /// create_batch - Create a new batch
```

Comment:
Passing BatchUpdates by reference doesn’t make much sense here, especially if the function is expected to mutate its fields (e.g., updated_at for audit purposes). It would be better to pass it by value to allow internal modifications without borrowing issues.
```suggestion
    async fn update_batch(&self, batch: &Batch, update: BatchUpdates) -> Result<Batch, DatabaseError>;
    /// create_batch - Create a new batch
```

Reply:
BatchUpdates only contains two fields
`pub struct BatchUpdates {
    pub end_block: u64,
    pub is_batch_ready: bool,
}`
We are not making any mutations in this inside `update_batch` function. I think it should be okay to pass it by reference? Or am I missing something?

Reply:
```
let mut updates = update.to_document()?;
```
I believe this is allowing the current implementation to work, 
either we remove this and let updates be mutable by itself, or we keep it as is.

Reply:
+1 @itsparser, let's fix this

Reply:
+1 @itsparser, let's fix this

Reply:
Done